ui/components/app/name/name-details/name-details.tsx

```diff
@@ -262,70 +272,85 @@ export default function NameDetails({
+            >
+              <Name
+                value={value}
+                type={NameType.ETHEREUM_ADDRESS}
+                disableEdit
+                internal
               />
-            }
-          />
-          <Label
-            flexDirection={FlexDirection.Column}
-            alignItems={AlignItems.flexStart}
-            marginBottom={2}
-            className="name-details__display-name"
-          >
-            {t('nameLabel')}
-            <FormComboField
-              hideDropdownIfNoOptions
-              value={name}
-              options={proposedNameOptions}
-              placeholder={t('nameSetPlaceholder')}
-              onChange={handleNameChange}
-              onOptionClick={handleProposedNameClick}
+            </div>
+            <Text marginBottom={4} justifyContent={JustifyContent.spaceBetween}>
+              {instructions}
+            </Text>
+            {/* @ts-ignore */}
+            <FormTextField
+              id="address"
+              className="name-details__address"
+              label={t('nameAddressLabel')}
+              value={formattedValue}
+              marginBottom={4}
+              disabled
+              endAccessory={
+                <ButtonIcon
+                  display={Display.Flex}
+                  iconName={
+                    copiedAddress ? IconName.CopySuccess : IconName.Copy
+                  }
+                  size={ButtonIconSize.Sm}
+                  onClick={handleCopyClick}
+                  color={IconColor.iconMuted}
+                  ariaLabel={t('copyAddress')}
+                />
+              }
             />
-          </Label>
-          <Button
-            variant={ButtonVariant.Primary}
-            startIconName={hasSavedName ? undefined : IconName.Save}
-            width={BlockSize.Full}
-            onClick={handleSaveClick}
-          >
-            {hasSavedName ? t('ok') : t('save')}
-          </Button>
+            <Label
+              flexDirection={FlexDirection.Column}
+              alignItems={AlignItems.flexStart}
+              marginBottom={2}
+              className="name-details__display-name"
+            >
+              {t('nameLabel')}
+              <FormComboField
+                hideDropdownIfNoOptions
+                value={name}
+                options={proposedNameOptions}
+                placeholder={t('nameSetPlaceholder')}
+                onChange={handleNameChange}
+                onOptionClick={handleProposedNameClick}
+              />
+            </Label>
+          </ModalBody>
+          <ModalFooter>
+            <Button
+              variant={ButtonVariant.Primary}
+              startIconName={IconName.Save}
+              width={BlockSize.Full}
+              onClick={handleSaveClick}
+              size={ButtonSize.Lg}
+            >
+              {t('save')}
+            </Button>
+          </ModalFooter>
         </ModalContent>
       </Modal>
     </Box>
```

Comment:
Excellent use of component library components here. It would be great to see a story for this component so it can be viewed in isolation for future design updates without having to traverse the code. Or guess the prerequisites of your account state to view.

Reply:
Hi George, some extra context here.

I explicitly didn't add stories for the `NameDetails` component originally since it's a nested component within the `Name` component. It has no usage beyond `Name` so it's really more of an extension to it or "private" in coding terms.

As a result, the various states of `NameDetails` largely overlap with `Name` hence why each `Name` story already has sufficient data to render the `NameDetails` modal when clicked and display suitable test data.

In addition to the duplication, there's also the additional complexity of rendering modals as flat in the page which means we wouldn't really be viewing them in their true form, or we'd need buttons to trigger them which would be similar to the current stories.

Apologies if we've a pattern to ideally make stories for all components, even nested ones. Just wanted to confirm it was considered but the gain seemed minimal in this scenario.